src/filesystem/impls/filer/FilerFileSystem.js

```diff
@@ -173,33 +173,31 @@ define(function (require, exports, module) {
             stat(path, callback);
         });
     }
-
-    function rename(oldPath, newPath, callback) {
+   function rename(oldPath, newPath, callback) {
         oldPath = decodePath(oldPath);
         newPath = decodePath(newPath);
 
         function updateBlobURL(err) {
             if(err) {
                 return callback(_mapError(err));
             }
-
+        fs.refresh(callback);
+        fs.reload();
             // If this was a rename on a file path, update the Blob cache too
-            stat(newPath, function(err, stat) {
+            /*stat(newPath, function(err, stat) {
```

Comment:
Why are you removing this?  We'll need it to figure out if we're on a dir or file.